jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererCodeCoverageTracker.kt

```diff
@@ -106,9 +108,16 @@ abstract class CodeWhispererCodeCoverageTracker(
     }
 
     private fun emitCodeWhispererCodeContribution() {
-        rangeMarkers.forEach {
-            if (!it.isValid) return@forEach
-            addAndGetAcceptedTokens(it.endOffset - it.startOffset)
+        rangeMarkers.forEach { rangeMarker ->
+            if (!rangeMarker.isValid) return@forEach
+            // if users add more code upon the recommendation generated from CodeWhisperer, we consider those added part as userToken but not CwsprTokens
+            val originalRecommendationLength = rangeMarker.getUserData(key)
+            originalRecommendationLength?.let {
+                addAndGetAcceptedTokens((rangeMarker.endOffset - rangeMarker.startOffset).coerceAtMost(originalRecommendationLength))
+            } ?: run {
```

Comment:
If user deletes some characters in the recommendation and add new characters back into the recommendations to make it as the same length as original, should we take account of the deletion but not the addition to the recommendation length, but here the logic would still treat it as recommendation staying the same.

Reply:
yea make sense, the other workaround I think of right now is to use what `usermodification` is using: Levenshtein edit distance, for example, we have the %modification on the recommendation generated by CWSPR, simply take (1 - %modification) and multiply by the original recomm length, then here we get unmodified accepted recommendation. Does this make sense to you @andrewyuq

Reply:
but I am still thinking if we have other approach to do this

Reply:
have some thoughts, let me check if it's doable and update here